js/src/views/Dapps/dappsStore.js

```diff
@@ -277,6 +291,12 @@ export default class DappsStore extends EventEmitter {
 
       this.apps = this.apps
         .filter((app) => !app.id || !newAppsIds.includes(app.id))
+        .filter((app) => { // check local dapp changes
+          if (app.type === 'local' && _local && apps.indexOf(app) === -1) {
```

Comment:
No, that filter is not designed to actually remove old apps, only add usable apps.. 

 don't want to rewrite this module for this, but for UI-2 I'll redo the module because ideally ALL apps should be removable and addable. But that's a PR for another time. This is minimally invasive.